docs/aws/audit/lambdamonitoring/rules/check_cross_account_access.mdx

```diff
@@ -23,6 +23,208 @@ HITRUST, SOC2, NISTCSF, PCIDSS
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Lambda functions in your AWS account. You can do this by using the AWS CLI command `aws lambda list-functions`. This command will return a list of all the Lambda functions in your AWS account.
+
+2. For each Lambda function, you need to get the resource policy. You can do this by using the AWS CLI command `aws lambda get-policy --function-name <function-name>`. Replace `<function-name>` with the name of your Lambda function. This command will return the resource policy of the Lambda function.
+
+3. Once you have the resource policy, you need to check if there are any statements that allow cross-account access. You can do this by looking for any statements where the `Principal` field is not equal to your AWS account ID or `*`.
+
+4. If you find any such statements, then the Lambda function allows cross-account access. If you don't find any such statements, then the Lambda function does not allow cross-account access. 
+
+Note: You can automate these steps by writing a Python script that uses the AWS SDK (boto3) to interact with the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary AWS SDK for Python (Boto3) modules in your Python script. You will need the 'boto3' module to interact with AWS services and the 'json' module to parse the AWS Lambda function policy.
+
+```python
+import boto3
+import json
+```
+
+2. Create a session using your AWS credentials. Replace 'aws_access_key_id', 'aws_secret_access_key', and 'aws_session_token' with your actual AWS credentials.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    aws_session_token='SESSION_TOKEN',
+)
+```
+
+3. Create a client for AWS Lambda.
+
+```python
+lambda_client = session.client('lambda')
+```
+
+4. List all the Lambda functions in your account and retrieve their policies. Check if the 'Principal' in the policy is set to '*' or contains an AWS account ID that is not yours. If it does, then the Lambda function allows cross-account access.
+
+```python
+functions = lambda_client.list_functions()
+for function in functions['Functions']:
+    policy = lambda_client.get_policy(FunctionName=function['FunctionName'])
+    policy_doc = json.loads(policy['Policy'])
+    for statement in policy_doc['Statement']:
+        if statement['Principal'] == '*' or statement['Principal'] != 'YOUR_AWS_ACCOUNT_ID':
+            print(f"Lambda function {function['FunctionName']} allows cross-account access.")
+```
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>